File: timbwa.cpp
#include <iostream>
using namespace std;

// Parent class
class Square
{
protected:
    double side;

public:
    Square(double s)
    {
        side = s;
    }

    ~Square() {}

    double getPerim()
    {
        return 4 * side;
    }

    double getArea()
    {
        return side * side;
    }
};

// Child class
class Cube : public Square
{
public:
    Cube(double s) : Square(s) {}

    ~Cube() {}

    double getArea()
    {
        return 6 * side * side;
    }

    double getVolume()
    {
        return side * side * side;
    }
};

int main()
{
    Cube c(5);

    cout << "Perimeter of base: " << c.getPerim() << endl;
    cout << "Surface area: " << c.getArea() << endl;
    cout << "Volume: " << c.getVolume() << endl;

    return 0;
}

File: timbwa.py
# Parent class
class Square:
    def __init__(self, side):
        self.side = side

    def getPeri(self):
        return 4 * self.side

    def getArea(self):
        return self.side * self.side


# Child class
class Cube(Square):
    def __init__(self, side):
        super().__init__(side)

    def getArea(self):
        return 6 * self.side * self.side

    def getVolume(self):
        return self.side * self.side * self.side


c = Cube(5)

print("Perimeter of base:", c.getPeri())
print("Surface area:", c.getArea())
print("Volume:", c.getVolume())
